agent-control/tests/on_host/oci_registry.rs

```diff
@@ -1,24 +1,212 @@
+
+    let mut blob_data = Vec::new();
+    file.read_to_end(&mut blob_data).await?;
+
+    let mut annotations: BTreeMap<String, String> = BTreeMap::new();
+    annotations.insert(
+        annotations::ORG_OPENCONTAINERS_IMAGE_TITLE.to_string(),
+        artifact.to_string(),
+    );
+
+    let layers = vec![ImageLayer::new(
+        blob_data.clone(),
+        manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE.to_string(),
+        Some(annotations),
+    )];
+
+    let config = Config {
+        data: blob_data,
+        media_type: manifest::IMAGE_CONFIG_MEDIA_TYPE.to_string(),
+        annotations: None,
+    };
+
+    let image_manifest = manifest::OciImageManifest::build(&layers, &config, None);
+
+    let _ = oci_client
+        .push(
+            reference,
+            &layers,
+            config,
+            &RegistryAuth::Anonymous,
+            Some(image_manifest),
+        )
+        .await
+        .map(|push_response| push_response.manifest_url)?;
+
+    fetch_manifest_and_get_digest(oci_client, reference).await
+}
+
+async fn fetch_manifest_and_get_digest(
+    oci_client: Client,
+    reference: &Reference,
+) -> Result<String, Box<dyn Error>> {
+    let (manifest, _) = oci_client
+        .pull_image_manifest(reference, &RegistryAuth::Anonymous)
+        .await?;
+
+    // Iterate over layers to find the one with the specified title annotation
+    for layer in manifest.layers {
+        if let Some(annotations) = &layer.annotations
+            && let Some(_layer_title) = annotations.get("org.opencontainers.image.title")
+        {
+            return Ok(layer.digest.clone());
+        }
+    }
+
+    Err(Box::new(DigestNotfoundError(
+        "Digest for artifact not found".to_string(),
+    )))
+}
+
+fn runtime_and_run_tag() -> (Arc<Runtime>, String) {
+    let runtime = Arc::new(
+        tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap(),
+    );
+
+    let rng = rand::rng();
+    let run_tag: String = rng
+        .sample_iter(&Alphanumeric)
+        .take(7)
+        .map(char::from)
+        .collect();
+    (runtime, run_tag)
+}
 
 #[test]
 #[ignore = "needs oci registry"]
-fn test_oci_registry_is_running_with_oci_registry() {
-    let registry_url = "http://localhost:5000".to_string();
-
-    let client = Client::builder()
-        .timeout(Duration::from_secs(5))
-        .build()
-        .expect("Failed to create HTTP client");
-
-    let response = client
-        .get(format!("{}/v2/", registry_url))
-        .send()
-        .expect("Failed to connect to OCI registry");
-
-    assert!(
-        response.status().is_success(),
-        "OCI registry should return 200 for /v2/ endpoint, got: {}",
-        response.status()
+fn test_download_artifact_from_local_registry_with_oci_registry() {
+    let (runtime, run_tag) = runtime_and_run_tag();
+
+    let reference = Reference::try_from(format!("localhost:5000/test:{}", run_tag)).unwrap();
```

Comment:
`localhost:5000` is the outside registry we set up for the test? if so could we make it a constant with some description? perhaps we can create a mod inside the common mod that wraps all things related to this OCI and some heleprs

Reply:
Yes, I am indeed using it in other parts of the test REGISTRY_URL but I forgot this one, thanks.